settings.gradle.kts

```
@@ -64,6 +66,38 @@ develocity {
   }
 }
 
+buildCache {
+  local {
+    // Local cache is enabled by default, but we might want to disable it in CI
+    // when using remote cache to avoid unnecessary storage
+    isEnabled = System.getenv("CI").isNullOrEmpty()
+  }
+  
+  remote<com.github.burrunan.s3cache.AwsS3BuildCache> {
+    region = "us-phoenix-1"
+    bucket = "opentelemetry-java-instrumentation-build-cache"
+    endpoint = "https://objectstorage.us-phoenix-1.oraclecloud.com"
+    
+    // Configure credentials for write access (only on main branch)
+    val accessKeyId = System.getenv("S3_BUILD_CACHE_ACCESS_KEY_ID")
+    val secretAccessKey = System.getenv("S3_BUILD_CACHE_SECRET_ACCESS_KEY")
+    
+    if (!accessKeyId.isNullOrEmpty() && !secretAccessKey.isNullOrEmpty()) {
+      awsAccessKeyId = accessKeyId
+      awsSecretKey = secretAccessKey
+      
+      // Enable push only for main branch builds
+      isPush = System.getenv("GITHUB_REF") == "refs/heads/main"
```

want it to be enabled whenever the secrets are passed
```suggestion
      isPush = true
```